imagedephi/redact/redact.py

```diff
@@ -173,6 +214,26 @@ def redact_images(
                 )
                 if output_file_counter == output_file_max:
                     logger.info("Redactions completed")
+                    if failed_img_counter:
+                        # Ensure that the logged index is the correct starting point
+                        with open(failed_manifest_file, "a") as manifest:
+                            yaml.dump(
+                                failed_images,
+                                manifest,
+                                explicit_start=True,
+                                default_flow_style=False,
+                            )
+                            manifest.write("failed_images_count: " + str(failed_img_counter) + "\n")
+                            index += 1
+                            yaml_command = f"""command: >
```

Comment:
This renders oddly and isn't quite a valid command.  I get
```
command: imagedephi run /tmp/a/Failed_2024-10-10_15-24-43\ --output-dir /tmp/a/Redacted_2024-10-10_15-24-43\
  --index 25\
```
in the yaml file.  The first `\` makes the input directory contain the following space.  Probably we should jam this all on one line rather than try to break it up (it would make it easier to copy to repeart).

Further, this creates a subfolder in the original redaction folder.  Either we need a flag to not put redactions in a subfolder or we shouldn't include the final segment of the path in the output directory.

This will also miss other command line options (--profile, --override-rules, --skip-rename, --recursive all probably matter).

Reply:
We can pull the `--profile` and `--skip-rename` from the original command. I think on this first pass, `--recursive` would be irrelevant because we are copying the failed images over into the top level of the 'failed directory'.

We wouldn't know what `--override-rules` would be used. We could include it in the command as `--override-rules [path-to-rules]` or some such thing.

Reply:
If the command specified `--recursive` and there are two files that we can't redact that have the same name but are in different subdirectories, what happens?  They would collide in the Failed folder.  Rerunning on the failed images won't reconstruct the recursive folder structure if we were using --skip-rename and --recursive.

I think we want to copy whatever `--override-rules` file was specified.  Specifically, I could imagine that I run my set on a bunch of files and it failed on two of them.  I then tweak my rules file and run it again with the new-and-improved rule file.

Reply:
That makes sense. I'll add this and the recursive behavior too.

Reply:
> in the yaml file. The first \ makes the input directory contain the following space. Probably we should jam this all on one line rather than try to break it up (it would make it easier to copy to repeart).

> Further, this creates a subfolder in the original redaction folder. Either we need a flag to not put redactions in a subfolder or we shouldn't include the final segment of the path in the output directory.

fixed in 140a339423c7c9af2c5e671acd597545c170de88

Reply:
> This will also miss other command line options (--profile, --override-rules, --skip-rename, --recursive all probably matter).

added in 7634ad8d05219b139150723774ca01788355ca3f

Reply:
> If the command specified --recursive and there are two files that we can't redact that have the same name but are in different subdirectories, what happens? They would collide in the Failed folder. Rerunning on the failed images won't reconstruct the recursive folder structure if we were using --skip-rename and --recursive.

fixed in 9eb13dca3d9d41b55ab72e333b9490d32023e12d